From deb10a4a710c3eade1c8077e331b865767a0a2d4 Mon Sep 17 00:00:00 2001
From: Liam Girdwood <lrg@slimlogic.co.uk>
Date: Sun, 13 Feb 2011 22:39:53 -0600
Subject: [PATCH 06/60] OMAP4: DMIC: Add DMIC codec platform device

commit 71262410dec17c1de15905fb1527b31d1d043598

Add platform device for DMIC codec interfaces available
in OMAP4.

Change-Id: I62d641f4dd3c898d10b9b50bb33c070837b24eaf
Signed-off-by: Liam Girdwood <lrg@slimlogic.co.uk>
Signed-off-by: Misael Lopez Cruz <misael.lopez@ti.com>
Integrated-by: Jingdong Lu <jingdong.lu@windriver.com>
---
 arch/arm/mach-omap2/devices.c |   18 ++++++++++++++++++
 1 files changed, 18 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/devices.c b/arch/arm/mach-omap2/devices.c
index cbc471c..4ea6396 100644
--- a/arch/arm/mach-omap2/devices.c
+++ b/arch/arm/mach-omap2/devices.c
@@ -294,6 +294,21 @@ static inline void omap_init_sti(void) {}
 
 #if defined CONFIG_ARCH_OMAP4
 
+static struct platform_device codec_dmic0 = {
+	.name	= "dmic-codec",
+	.id	= 0,
+};
+
+static struct platform_device codec_dmic1 = {
+	.name	= "dmic-codec",
+	.id	= 1,
+};
+
+static struct platform_device codec_dmic2 = {
+	.name	= "dmic-codec",
+	.id	= 2,
+};
+
 static struct platform_device omap_abe_dai = {
 	.name	= "omap-abe-dai",
 	.id	= -1,
@@ -301,6 +316,9 @@ static struct platform_device omap_abe_dai = {
 
 static inline void omap_init_abe(void)
 {
+	platform_device_register(&codec_dmic0);
+	platform_device_register(&codec_dmic1);
+	platform_device_register(&codec_dmic2);
 	platform_device_register(&omap_abe_dai);
 }
 #else
-- 
1.7.4.1

